src/packages/engine-core/src/NodeEngine.ts

```diff
@@ -194,17 +198,18 @@ export class NodeEngine implements Engine {
         )}\` were removed, you can now safely remove them from your schema.prisma.`,
       )
     }
-
     this.previewFeatures = this.previewFeatures.filter(
       (e) => !removedFlags.includes(e),
     )
+    
+    // custom endpoint
```

Comment:
What is this exactly?

Reply:
Looks like it's not tested nor used.

From this
```
if (engineEndpoint) {
  const url = new URL(engineEndpoint)
  this.port = Number(url.port)
}
```

It looks like engineEndpoint needs to be a valid URL and then it parses the port and uses that port to communicate with the engine.

So a feature to tell the client to use an already existing engine on a defined port?

Reply:
But only the port, not the host (which would make this a way to talk to a hosted Engine running somewhere else maybe)?

Reply:
Indeed.

Actually, it seems that Tim introduced it for Dom:

Context https://prisma-company.slack.com/archives/CEYCG2MCN/p1592992017273200?thread_ts=1592985426.262900&cid=CEYCG2MCN

>In 2.2.0-dev.2, we now have a new way to override the engine:

```
  const client = new PrismaClient({
    __internal: {
      engine: {
        endpoint: `http://localhost:12345`
      }
    }
  } as any)
```

>It’s mostly for Dom for now, so he can start the engine himself, but it might come handy if you need to debug. With that you can now do things like this:
```
  const client1 = new PrismaClient()
  await client1.connect()
  const port = (client1 as any).engine.port
  const engineEndpoint = `http://localhost:${port}`
  const client2 = new PrismaClient({
    __internal: {
      engine: {
        endpoint: engineEndpoint
      }
    }
  } as any)
  const users = await client2.user.findMany()
  console.log({ users })
```

Reply:
```suggestion
    // internal only: custom endpoint (see https://prisma-company.slack.com/archives/CEYCG2MCN/p1592992017273200?thread_ts=1592985426.262900&cid=CEYCG2MCN)
```